packages/es-components/src/components/patterns/datepicker/DatePicker.js

```diff
@@ -30,25 +30,55 @@ class DateTextbox extends React.Component {
   }
 }
 
-const NativeDatePicker = props => {
-  const onChangeIntercept = event => props.onChange(moment(event.target.value));
+const getVerifiedDate = selectedDate => {
+  let verifiedDate = selectedDate;
+  if (typeof selectedDate === 'string') {
+    const newDateMoment = moment(selectedDate, 'L');
+    const isFullDate =
+      selectedDate && newDateMoment.format('L') === selectedDate;
+    verifiedDate = isFullDate ? newDateMoment : undefined;
+  }
+  return verifiedDate;
+};
+
+const NativeDatePicker = ({
+  selectedDate,
+  name,
+  onChange,
+  onChangeRaw,
+  ...props
+}) => {
+  const onChangeIntercept = event => onChange(moment(event.target.value));
+  const onChangeRawIntercept = event =>
+    (onChangeRaw || noop)(moment(event.target.value));
+  const verifiedDate = getVerifiedDate(selectedDate);
   const dateValue =
-    !!props.selectedDate && props.selectedDate.isValid()
-      ? props.selectedDate.format('YYYY-MM-DD')
+    !!verifiedDate && verifiedDate.isValid()
+      ? verifiedDate.format('YYYY-MM-DD')
       : '';
 
   return (
     <DateTextbox
-      name={props.name}
+      name={name}
       prependIconName="calendar"
       type="date"
       value={dateValue}
       {...props}
       onChange={onChangeIntercept}
+      onChangeRaw={onChangeRawIntercept}
```

Comment:
I don't think the `DateTextbox` will do anything with this `onChangeRaw` prop, unless I'm missing something? It's just an input field under the covers, and I don't see us doing anything with `onChangeRaw` in the `Textbox` component. Again though, I could be wrong, so correct me if I am!

If I'm not, and we really want this raw behavior, we could probably do something with the `onChangeIntercept` we're creating. Call `onChangeRaw` for any change, but only call `onChange` if the date is valid, or something to that effect.

Reply:
I can double check this. I was seeing some weird behavior with the native datepicker, and this seemed to resolve it.

Reply:
onChangeRaw is a prop used by the react-datepicker to track the actual text value of the input rather than a moment value. I don't think this is needed here.

Reply:
Looks like it wasn't necessary. I'm not sure what fixed my issue, but I've removed this.